### PR TITLE
sunpy.sun python file review

### PR DESCRIPTION
**Pull request by nabobalis**
This time for `sun`.

TODO:

- [ ] longitude_Sun_perigee, true_latitude and apparent_latitude need to be fixed
- [x] Add breaking change log.

### CODE REVIEW

File: sunpy/sun/_constants.py [hunk 1]
```diff
@@ -1,21 +1,11 @@
...
-
+# We want to keep the old constants for now.
+# TODO: Need better sources for some constants as well as error values.
 # This is to keep the old solar values until we make the choice to move to IAU2015 Values
-import astropy
-if int(astropy.__version__[0]) >= 2:
-    import astropy.constants.astropyconst13 as astrocon
-else:
-    import astropy.constants as astrocon
+import astropy.constants.astropyconst13 as astrocon
```
Comment (Cadair) [inline sunpy/sun/_constants.py]: We should update this for 1.0.

File: setup.cfg [hunk 2]
```diff
@@ -48,7 +48,7 @@ docs =
   ruamel.yaml
   sphinx
   sphinx-astropy
-  sphinx-gallery
+  sphinx-gallery<0.3.0
```
Comment (Cadair) [inline setup.cfg]: remove this apparently

File: sunpy/sun/sun.py [hunk 3]
```diff
@@ -21,156 +17,171 @@
...
     """
     ra = true_rightascension(t)
     dec = true_declination(t)
     return ra, dec
 
 
-@add_common_docstring(append=PARAMETER_DOCS)
-def eccentricity_SunEarth_orbit(t='now'):
+@add_common_docstring(**_variables_for_parse_time_docstring())
+def eccentricity_sunearth_orbit(t='now'):
```
Comment (Cadair) [inline sunpy/sun/sun.py]: Wouldn't `eccentricity_sun_earth_orbit` be more appropriate?

File: sunpy/sun/sun.py [hunk 4]
```diff
@@ -74,7 +74,7 @@ def position(t='now'):
 
 
 @add_common_docstring(**_variables_for_parse_time_docstring())
-def eccentricity_sunearth_orbit(t='now'):
+def eccentricity_sun_earth_orbit(t='now'):
```
Comment (Cadair) [inline sunpy/sun/sun.py]: The next question becomes should we provide a deprecated wrapper for this?

Comment (nabobalis) [reply sunpy/sun/sun.py]: Fair question, I did add the breaking changelog just in case but maybe its not enough

**Pull request merged**